app/services/upload-knowledge.js

```diff
@@ -0,0 +1,52 @@
+const { uploadKnowledge, updateKnowledgeMetadata } = require('../api/knowledge')
+const { mime } = require('../constants/document-types')
+const { getExtension } = require('../lib/file')
+
+const getBuffer = (payload) => {
+  let buffer
+  let ext
+
+  if (payload.uploadtype === 'file') {
+    buffer = payload.document._data
+    ext = getExtension(payload.document.hapi.filename)
+  } else {
+    buffer = Buffer.from(payload.usertext)
+    ext = 'txt'
+  }
+
+  const type = mime[ext]
+
+  return { buffer, type }
+}
+
+const buildMetadataPayload = (payload) => {
+  const fileName = payload.uploadtype === 'file'
+    ? payload.document.hapi.filename
+    : 'txt'
+
+  return {
+    fileName,
+    title: payload.title,
+    category: payload.category,
+    source: 'frontend'
+  }
+}
+
+const upload = async (payload) => {
+  const { buffer, type } = getBuffer(payload)
+
+  const { id } = await uploadKnowledge(buffer, type)
+
+  const metadataPayload = buildMetadataPayload(payload)
+
+  await updateKnowledgeMetadata(id, metadataPayload)
+}
+
+const saveUri = async (payload) => {
```

Comment:
Probably should be avoiding creating placeholder functions for now. We'll come back to this once we have a plan in place.